### PR TITLE
GOD-205: Better exception msging from godmother->mobileapi

### PR DESCRIPTION
**Pull request by zain-cheng**

### CODE REVIEW

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java [hunk 1]
```diff
@@ -260,12 +265,54 @@ private void setStatusCode(ContainerRequestContext requestContext, ContainerResp
...
+		Throwable throwable = (Throwable) entity;
+		String statusCodeAsString = String.valueOf(responseContext.getStatus());
+		String reasonPhrase = responseContext.getStatusInfo() != null ?
+													responseContext.getStatusInfo().getReasonPhrase() : DEFAULT_STATUS.getReasonPhrase();
+		String applicationMessage = throwable.getCause() != null ?
+																throwable.getCause().getMessage() : throwable.getMessage();
+
+		Error error = new Error();
+		error.setStatus(statusCodeAsString); // Numerical status code, expressed as a String
+		error.setCode(statusCodeAsString); 		// Application-specific error code, expressed as a String
```
Comment (cfries3) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: These two fields are getting set to the same thing, do we need that duplication?

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java [hunk 2]
```diff
@@ -260,12 +265,54 @@ private void setStatusCode(ContainerRequestContext requestContext, ContainerResp
...
+	private void setErrorEntity(ContainerResponseContext responseContext) {
+
+		Object entity = responseContext.getEntity();
+		assert(entity!=null && entity instanceof Throwable);
+
+		Throwable throwable = (Throwable) entity;
+		String statusCodeAsString = String.valueOf(responseContext.getStatus());
+		String reasonPhrase = responseContext.getStatusInfo() != null ?
+													responseContext.getStatusInfo().getReasonPhrase() : DEFAULT_STATUS.getReasonPhrase();
+		String applicationMessage = throwable.getCause() != null ?
```
Comment (cfries3) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: The spacing here is weirdly hard to read on github, whitespace fix?

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java (hunk 1 above)
Comment (cfries3) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: Nowhere in here is the name of the exception being thrown, whether it is a null pointer from godmother or a client exception from downstream, I feel that would be useful information, maybe set code to the reason phrase and title to the exception class name?

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java (hunk 2 above)
Comment (zain-cheng) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: yea, the spacing is weird.  It's just the default for the rest of the file, which is 2-spaced tabs.

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java (hunk 1 above)
Comment (zain-cheng) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: Yes, I was kinda conflicted as I was refactoring this. And thinking about it again, I feel that what you mentioned makes sense.  Making the change now.

Comment (zain-cheng) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: Just updated with the changes.

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java [hunk 3]
```diff
@@ -260,12 +267,57 @@ private void setStatusCode(ContainerRequestContext requestContext, ContainerResp
...
+		Object entity = responseContext.getEntity();
+		assert(entity!=null && entity instanceof Throwable);
+
+		Throwable throwable = (Throwable) entity;
+		String statusCodeAsString = String.valueOf(responseContext.getStatus());
+		String reasonPhrase = responseContext.getStatusInfo() != null ?
+													responseContext.getStatusInfo().getReasonPhrase() : DEFAULT_STATUS.getReasonPhrase();
+
+		String problemSummary = throwable.getCause() != null ?
+														throwable.getClass().getSimpleName() : throwable.getClass().getSimpleName();
```
Comment (cfries3) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: I think you meant thowable.getCause().getClass().getSimpleName : throwable.getClass().getSimpleName()

Comment (zain-cheng) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: oh nice catch. :+1:

Comment (zain-cheng) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: fixed.

**Comment by cfries3** [comment]
lgtm

**Pull request merged**